File: techievolve/public/js/customer.js
frappe.ui.form.on('Customer', {
    onload: function(frm){
        if (!frm.doc.__islocal) {
			frm.add_custom_button(__("Create Login"), function () {
                let dialog = new frappe.ui.Dialog({
                    'title': 'Create User',
                    'fields': [
                        { fieldtype: "Data", fieldname: "user_name", label: __('User Name'),reqd:1  },
                        { fieldtype: "Password", fieldname: "password", label: __('Password'),reqd:1 },
                    ],
                });
                dialog.show();
                dialog.set_primary_action(__('Create'), function () {
                var values = dialog.get_values();
                frappe.call({
                    method: "techievolve.techievolve.doc_events.customer.create_user",
                    args: {
                        email_id: frm.doc.email_id,
                        user_name: values.user_name,
                        password: values.password,
                        first_name: frm.doc.customer_name
                    },
                    callback: (r) => {
                        if (r.message) {
                            //frappe.msgprint('Customer changed successfully');
                            dialog.hide();
                            frm.reload_doc();
                        }
                    }
                })
                });
                dialog.get_close_btn().on('click', () => {
                    dialog.hide();
                });
            });
        }   
    }   
});

File: techievolve/techievolve/doc_events/customer.py
import frappe

@frappe.whitelist()
def create_user(email_id,user_name,password,first_name):
    if email_id and user_name and password:
        doc = frappe.new_doc("User")
        doc.email = email_id
        doc.username = user_name
        doc.new_password = password
        doc.first_name = first_name
        doc.append('roles',{
            'role': "Customer"
        })
        doc.save()
        return "User Created"
